### PR TITLE
update eslint-vue and prevent reactivity loss of props

### PR DESCRIPTION
**Pull request by Kr0nox**
This PR updates the eslint vue plugin to the newest version.

It also enables the newly added rule to prevent reactivity loss through prop deconstrcution. This rule was implemneted in all places violating it

Replaces #1245

### CODE REVIEW

**Review by sebinside — APPROVED** [review]

**Comment by Kr0nox** [comment]
@sebinside In all PR i will need to merge develop into the branch once. Should I wait for your approval on all those PRs again or can I just merge them?

**Comment by sebinside** [comment]
@Kr0nox As all of these PRs don't change bigger parts of the logic, please just go ahead and merge the PRs, thank you.

**Pull request merged**
